decidim-core/lib/decidim/core.rb

```diff
@@ -247,6 +247,16 @@ def self.seed!
     2.days
   end
 
+  # Whether to expose the GraphiQL playground, by default only in development and test
+  config_accessor :expose_graphiql do
+    !Rails.env.production?
```

Comment:
I think this should be opt out. Even if the installation decides to not enable this API it's possible to have this kind of Information Disclosure by seeing the source code that they're obligated to publish by the license. Also this is some kind of Security Through Obscurity and could even be against Social Contract (https://decidim.org/contract) on the Open Content and Transparency clauses.

Reply:
The issue this fixes is that it was enabled by default --this commit makes graphiql opt-in in production (by disabling it by default except when `Rails.env.production?`, which is a sane default).

I'm not sure what you mean in this context by making it opt-out?

Reply:
Just to clarify, this only disables Graph**i**QL (the embedded GraphQL client), not the whole GraphQL API

Reply:
Be aware that disabling graphiql will cause `api/docs` documentation to fail due this line as the method `graphiql_path` won't be available:
https://github.com/decidim/decidim/blob/a070c97872cd9d1ff305251aaea0ab27982318bd/decidim-api/app/views/decidim/api/documentation/show.html.erb#L3

Reply:
@microstudi since we implemented it as a route constraint (not mounting the route itself conditionally, but rather allowing access on a request per request basis), the `graphiql_path` method will generate the right route even if it's disabled, but trying to click there will raise a routing error.

In any case, good point because we shouldn't be linking there if it's not enabled --will amend this! Thank you.

Reply:
As requested by @andreslucena I made GraphiQL exposed by default, but opt-out.

Reply:
Rember to update the changelog then! thanks!

Reply:
I did @microstudi :) Thanks!